### ISSUE TITLE
"No Prefix Constructor Completions" error when using Context Information

### ISSUE THREAD

**Issue by snoobvn** [spring-projects/spring-ide#226]
Version: STS 3.9.1

>  The 'No Prefix Constructor Completions' proposal computer from the 'org.springsource.ide.eclipse.commons.completions' plug-in did not complete normally. The extension violated the API contract of the 'computeContextInformation()' operation.

 I have the same problem as https://github.com/spring-projects/spring-ide/issues/164

Context Assist (Ctrl+Space) works fine but not Context Information (Ctrl+Shift+Space)
![test](https://user-images.githubusercontent.com/6419729/33800226-631dc19c-dd6e-11e7-868c-e50caf6ca9a6.gif)

The error log doesn't provide anything useful:

>!ENTRY org.eclipse.jdt.ui 2 0 2017-12-10 05:41:33.385
!MESSAGE The 'No Prefix Constructor Completions' proposal computer from the 'org.springsource.ide.eclipse.commons.completions' plug-in did not complete normally. The extension violated the API contract of the 'computeContextInformation()' operation.

**Comment by martinlippert** [comment]
Can you attach a sample project that helps us to reproduce this problem? Since we improved the way it works behind the scenes for the 3.9.1 release, it looks like your issue might be caused by something else.

Also do you have anything installed into your STS in addition to the vanilla STS distro? Would be interesting to know.

**Comment by snoobvn** [comment]
Since I've changed the workspace and created new Java project but the problem still exists so I don't think it is releated to any specfic project.


[preferences.zip](https://github.com/spring-projects/spring-ide/files/1586104/preferences.zip)

Installed Plugins:
![plugins2](https://user-images.githubusercontent.com/6419729/34342588-28357a90-e9e7-11e7-8e5f-adf17b786157.png)
![plugins](https://user-images.githubusercontent.com/6419729/34342582-11dd138e-e9e7-11e7-8b70-038d5ea0c6e5.png)

**Comment by martinlippert** [comment]
@snoobvn Can you reproduce this problem with a vanilla and fresh STS 3.9.2 distribution? If so, can you post the steps to reproduce it?

**Comment by hanming-nls** [comment]
The following steps are what I have used to reproduced the problem.

1) download http://download.springsource.com/release/STS/3.9.2.RELEASE/dist/e4.7/spring-tool-suite-3.9.2.RELEASE-e4.7.2-win32-x86_64.zip

2) unzip to drive without whitespace

3) launch sts on a new workspace

4) create a new java project

5) create a new class com.test.Main
`
package com.test;
public class Main {
	public static void main(String[] args) {
		System.out.println("start");
		
		Main main = new Main();
		main.test();
		
		System.out.println("end");
	}
	
	void test(int a, int b) {
		test(a);
		test(b);
	}
	
	void test(int i) {
		test();
	}
	
	void test() {
		System.out.println("test");
	}
}
`

6) as the test method is overloaded, go to any of the test( {cursor} ) line and and issue ctrl+shift+space
![image](https://user-images.githubusercontent.com/33014397/34978952-e001bbc6-fada-11e7-8175-ffb40dbfae15.png)

**Comment by martinlippert** [comment]
Thanks a lot for the detailed description. Works nicely on my machine, but that doesn't really mean anything here, since the error message indicates a timing constraint and that might heavily depend on the actual machine you are running this on.

But anyway, since this seems to reproduce the problem for you even with a fresh STS install and an extremely simple workspace, it is a good indicator for where the problem might be, so still of great value to us. Thanks again.

Will take a deeper look to see what we can do here.

**Comment by hanming-nls** [comment]
Not sure if it helps to mention that I'm behind a proxy that block certain sites since I relate "timing constraint" to a possibility of "timeout"

As I switch around the network often, I do seem to have some initialization problem occasionally on the my primary sts, which is updated to 3.9.2 from earlier version, if I didn't have the proper connection upon startup. 

And, lastly, I'm running on a Windows 2016 server which have more security settings enabled by default.

Do feel free to let me know if you need more information.

**Comment by martinlippert** [comment]
I am really really sorry about this. I worked on this today for a while (added a safety timeout for the operation to not cause any hiccups with the JDT completion processor timeouts, which is a great safety improvement), but then I figured out that (after reading the initial bug report description very carefully again) that this bug is describing something totally different. And to my bad, the bug that is described here is totally obvious - and could get fixed within a minute.

So I did, this is fixed now, CI builds for this are running.

I have to apologize again for me overseeing the details in the description and since then keeping this in mind in the wrong way. Sorry again, hope the fix that I implemented now works for you all.

And thanks again for reporting this and reporting the details.
